h/services/email.py

```diff
@@ -18,22 +19,35 @@
 DAILY_SENDER_MENTION_LIMIT = 100
 
 
-@dataclass(frozen=True)
+@dataclass
 class EmailData:
     recipients: list[str]
     subject: str
     body: str
     tag: EmailTag
     html: str | None = None
     subaccount: str | None = None
+    reply_to: str | None = None
+    from_name: str | None = None
 
-    @property
-    def message(self) -> pyramid_mailer.message.Message:
+    def message(self, default_sender: str) -> pyramid_mailer.message.Message:
         extra_headers: dict[str, str] = {"X-MC-Tags": self.tag}
+
         if self.subaccount:
             extra_headers["X-MC-Subaccount"] = self.subaccount
+
+        if self.reply_to:
+            extra_headers["Reply-To"] = self.reply_to
```

Comment:
When sending an email (any email) if `EmailData.reply_to` was set then use it as the email's `Reply-To` header.